### PR TITLE
Add reliquary support as staking option & calculate apr

### LINKED ISSUE #154: Add support for reliquary as a staking option & calculate apr
We need to support reliquary as a staking and calculate apr's based on it
- staking service
- apr calculator

### PR DESCRIPTION
**Pull request by 0xSkly**
resolves #154

### CODE REVIEW

File: modules/config/network-config.ts
Comment (franzns) [inline modules/config/network-config.ts]: remove unused

File: modules/config/network-config.ts
Comment (franzns) [inline modules/config/network-config.ts]: remove

File: modules/pool/lib/apr-data-sources/fantom/reliquary-farm-apr.service.ts
Comment (franzns) [inline modules/pool/lib/apr-data-sources/fantom/reliquary-farm-apr.service.ts]: typo: detail

File: modules/pool/lib/apr-data-sources/fantom/reliquary-farm-apr.service.ts
Comment (franzns) [inline modules/pool/lib/apr-data-sources/fantom/reliquary-farm-apr.service.ts]: remove, unused

File: modules/config/network-config.ts
Comment (franzns) [inline modules/config/network-config.ts]: make optional

File: modules/pool/lib/staking/fantom/reliquary-staking.service.ts
Comment (franzns) [inline modules/pool/lib/staking/fantom/reliquary-staking.service.ts]: allocation points never change for a level, do they?

File: modules/pool/lib/staking/fantom/reliquary-staking.service.ts
Comment (franzns) [inline modules/pool/lib/staking/fantom/reliquary-staking.service.ts]: remove, unused

File: modules/pool/pool.prisma
Comment (franzns) [inline modules/pool/pool.prisma]: should we remove that as well?

Comment (franzns) [inline modules/pool/pool.prisma]: should we also remove this?

File: modules/subgraphs/reliquary-subgraph/reliquary.service.ts
Comment (franzns) [inline modules/subgraphs/reliquary-subgraph/reliquary.service.ts]: I think this has the limit of 6k users (max skip). Might want to implement a paged approach to get all users.

**Review by franzns — COMMENTED** [review]
Few minors things. Looks good, we should give it a try. 

It's still missing any crons and integration into the pool service.

File: modules/subgraphs/reliquary-subgraph/reliquary.service.ts
Comment (franzns) [inline modules/subgraphs/reliquary-subgraph/reliquary.service.ts:54]: this only works for up to 6k relics

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (franzns) [inline modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts]: should be reliquary address

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (franzns) [inline modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts]: that could also match on a mastechef farm id?
need to only get pools with reliquary staking type

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (franzns) [inline modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts]: need to add `reliquary-` in front of the id string

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (franzns) [inline modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts]: id string should be `reliquary-${relic.pid}-${relic.userAddress}`

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (franzns) [inline modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts:170]: In theory, a pool could have a farm both on masterchef and on reliquary. I think that causes a problem down the line. 

Any way we can make sure this doesn't happen?

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (franzns) [inline modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts]: Doesn't work if a user has more than one relic for the same pool. Also need to add all amounts.

**Review by franzns — CHANGES_REQUESTED** [review]
Few things need to change

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (0xSkly) [reply modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts]: true, good catch!

Comment (0xSkly) [reply modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts]: also added this in the master chef version

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (0xSkly) [reply modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts]: you mean in the create part right?

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (0xSkly) [reply modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts]: correct

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (0xSkly) [reply modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts:170]: mh from what i see the resolver would just pass down both staking options, farm (masterxhef) & reliquary in the `GqlPoolStaking` type. But then it should never be that both have emissions and i think this can only really happen when we migrate the other pools beside fbeets since fbeets gets a new pool anyways. So we'd have to clean up the database when we do the migration but think there is quitet some more stuff to think about when we do this

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (0xSkly) [reply modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts]: yea mixed some thoughts here, this should be better i think :)

File: modules/pool/lib/staking/fantom/reliquary-staking.service.ts
Comment (0xSkly) [reply modules/pool/lib/staking/fantom/reliquary-staking.service.ts]: true, thought they can but nope

File: modules/pool/pool.prisma
Comment (0xSkly) [reply modules/pool/pool.prisma]: removed it

File: modules/subgraphs/reliquary-subgraph/reliquary.service.ts
Comment (0xSkly) [reply modules/subgraphs/reliquary-subgraph/reliquary.service.ts]: removed it since its not being used

File: modules/subgraphs/reliquary-subgraph/reliquary.service.ts
Comment (0xSkly) [reply modules/subgraphs/reliquary-subgraph/reliquary.service.ts:54]: true, took the approach as in the masterchef staking balance service

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (franzns) [inline modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts]: shouldn't this be `entryTimestamp_gt` and not `gte` ? Like this you'll get duplicates for the timestamp since they were already in the last batch.

**Review by franzns — APPROVED** [review]

**Comment by franzns** [comment]
You also need to hook up the APR service.

File: modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts
Comment (0xSkly) [reply modules/user/lib/fantom/user-sync-reliquary-farm-balance.service.ts]: using relicId

**Comment by 0xSkly** [comment]
hooked up the apr service, will run some local tests anyways before we merge but kinda blocked until we can deploy the new version of reliquary

**Pull request merged**
